eox_nelp/pearson_vue/api/v1/views.py

```diff
@@ -0,0 +1,158 @@
+"""
+Views for handling Pearson VUE request.
+
+This module defines views for handling various types of request in the Pearson VUE service.
+
+Classes:
+    PearsonRTENBaseView: A base view class for creating Pearson VUE RTEN events.
+    ResultNotificationView: A view for handling result notification events.
+    PlaceHoldView: A view for handling place hold events.
+    ReleaseHoldView: A view for handling release hold events.
+    ModifyResultStatusView: A view for handling modify result status events.
+    RevokeResultView: A view for handling revoke result events.
+    UnrevokeResultView: A view for handling unrevoke result events.
+"""
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import generics, status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from eox_nelp.pearson_vue.api.v1.serializers import PearsonRTENSerializer
+from eox_nelp.pearson_vue.constants import (
+    MODIFY_RESULT_STATUS,
+    PLACE_HOLD,
+    RELEASE_HOLD,
+    RESULT_NOTIFICATION,
+    REVOKE_RESULT,
+    UNREVOKE_RESULT,
+)
+from eox_nelp.pearson_vue.models import PearsonRTENModel
+
+
+class PearsonRTENBaseView(generics.ListCreateAPIView):
+    """
+    Base view for Pearson RTEN (Real Time Event Notification) API endpoints.
+
+    This view provides the base functionality for creating Pearson RTEN events.
+    Subclasses should define the `event_type` attribute to specify the type of event.
+
+    Attributes:
+        permission_classes (list): List of permission classes required for the view.
+        serializer_class (Serializer): Serializer class used for validating and parsing data.
+        queryset (QuerySet): Queryset representing the model instances to operate on.
+        authentication_classes (list): List of authentication classes used for authentication.
+    """
+    event_type = None
+    permission_classes = [IsAuthenticated]
+    serializer_class = PearsonRTENSerializer
+    queryset = PearsonRTENModel.objects.all()  # pylint: disable=no-member
+    authentication_classes = [JwtAuthentication]
```

Comment:
What do you think of adding session authentication. And staff or admin permission?. This to allow check the API via webrowser?
![image](https://github.com/eduNEXT/eox-nelp/assets/51926076/158ed979-2e75-4e8b-bbb5-2c60ca0f2776)

Reply:
That requires  that the credentials that we will share with Pearson are associated with an staff or admin user, and we won't give admin credentials to them.

on the other hand, the session authentication doesn't make sense for me, since this depends on a certificate, probably I will  do this more restrictive and include the Bearer authentication that I found in eox-tenant